Gems/Multiplayer/Code/Source/Pipeline/NetworkPrefabProcessor.cpp

```diff
@@ -53,143 +81,108 @@ namespace Multiplayer
-        {
-            Instance* instance = netEntityToInstanceMap[prefabEntity];
-            
-            AZ::EntityId entityId = prefabEntity->GetId();
-            AZ::Entity* netEntity = instance->DetachEntity(entityId).release();
-            AZ_Assert(netEntity, "Unable to detach entity %s [%s] from the source prefab instance", 
-                prefabEntity->GetName().c_str(), entityId.ToString().c_str());
-
-            netEntity->InvalidateDependencies();
-            netEntity->EvaluateDependencies();
-
-            auto* transformComponent = netEntity->FindComponent<AzFramework::TransformComponent>();
-            if (transformComponent)
-            {
-                AZ::EntityId parentId = transformComponent->GetParentId();
-                if (parentId.IsValid() && !prefabNetEntityIds.contains(parentId))
-                {
-                    // Clear parent ID for net entities parented to a non-net entity.
-                    // To be addressed by the spawnable aliases system where non-net entities
-                    // will be spawned together with the networked ones in which case we'll keep
-                    // the cross-spawnable references.
-                    transformComponent->SetParent(AZ::EntityId());
-                }
-            }
-
-            prefabNetEntityIds.insert(netEntity->GetId());
+        // Grab all net entities with their corresponding Instances to handle nested prefabs correctly
+        PopulateNetworkInstance(
+            &sourceInstance,
+            &networkInstance,
+            &sourceInstance,
+            &networkInstance,
+            prefab.GetName(),
+            prefabName,
+            context,
+            networkSpawnableAssetId);
 
-            // Insert the entity into the target net spawnable
-            netSpawnableEntities.emplace_back(netEntity);
-        }
+        context.AddPrefab(AZStd::move(networkPrefab));
 
-        // Add net spawnable asset holder to the prefab root
-        {
-            EntityOptionalReference containerEntityRef = sourceInstance.GetContainerEntity();
-            if (containerEntityRef.has_value())
-            {
-                auto* networkSpawnableHolderComponent = containerEntityRef.value().get().CreateComponent<NetworkSpawnableHolderComponent>();
-                networkSpawnableHolderComponent->SetNetworkSpawnableAsset(networkSpawnableAsset);
-            }
-            else
-            {
-                AZ::Entity* networkSpawnableHolderEntity = aznew AZ::Entity(uniqueName);
-                auto* networkSpawnableHolderComponent = networkSpawnableHolderEntity->CreateComponent<NetworkSpawnableHolderComponent>();
-                networkSpawnableHolderComponent->SetNetworkSpawnableAsset(networkSpawnableAsset);
-                sourceInstance.AddEntity(*networkSpawnableHolderEntity);
-            }
-        }
+        m_processedNetworkPrefabs.insert(prefabName);
 
-        context.GetProcessedObjects().push_back(AZStd::move(object));
```

Comment:
The removal of this implies that the context will no longer have products to return. Is this okay? @SergeyAMZN @srikappa-amzn, unsure if either of you have thoughts here.

Reply:
Yes, that's correct. The prefab processor now creates a **prefab**, not a spawnable product. 
Spawnable products are created by the PrefabCatchmentProcessor